cmd/libsnap-confine-private/infofile.c

```diff
@@ -76,6 +82,31 @@ int sc_infofile_get_key(FILE *stream, const char *key, char **value, sc_error **
         }
         /* Replace the trailing newline character with the NUL byte. */
         line_buf[nread - 1] = '\0';
+
+        /* Handle ini sections (if requested via non-null section name) */
+        if (section != NULL) {
+            if (line_buf[0] == '[') {
+                section_matched = false;
+                char *sec_endptr = memchr(line_buf, ']', nread);
+                if (sec_endptr == NULL) {
+                    err = sc_error_init_simple("line %d is not a valid ini section", lineno);
+                    goto out;
+                }
+                /* Replace closing ']' with string terminator byte */
+                *sec_endptr = '\0';
+                const char *scanned_section = line_buf + 1;
+                if (sc_streq(scanned_section, section)) {
```

Comment:
hm for input such as:
```
[]
foo=bar
```
we will find a section `""` and assume it's valid. Perhaps the could should explicitly complain if `section != NULL && strlen(section) == 0` somewhere at the top of this function?

Reply:
Sounds reasonable, will do, but as a general remark we're not after a full fledged ini parser here anyway, e.g. we allow duplicated keys etc. In the same vein we don't care about duplicated sections.